daemon/internal/newrelic/integration/test.go

```diff
@@ -285,10 +289,15 @@ func ScrubHost(in []byte) []byte {
 	return re.ReplaceAll(in, []byte("__HOST__"))
 }
 
-func SubEnvVars(in []byte) []byte {
+func (t *Test) subEnvVars(in []byte) []byte {
 	re := regexp.MustCompile("ENV\\[.*?\\]")
 	return re.ReplaceAllFunc(in, func(match []byte) []byte {
-		return []byte(os.Getenv(string(match[4 : len(match)-1])))
+		k := string(match[4 : len(match)-1])
+		v, present := os.LookupEnv(k)
+		if !present {
+			v = t.Env[k]
```

Comment:
What happens if the key is not in in or the process environment variables? Does this return nil?